boomstream.py

```diff
@@ -26,27 +31,38 @@
   'sec-fetch-dest': 'document',
   'accept-language': 'en-US,en;q=0.9,ru;q=0.8,es;q=0.7,de;q=0.6'}
 
-class App():
+def valid_filename(s):
+    return ''.join(c for c in s if c in VALID_FILENAME_CHARS or c.isalpha())
+
+def output_path(path):
+    return os.path.join(OUTPUT_PATH, path)
+
+def ensure_folder_exists(path):
+    if not os.path.exists(path):
+        os.mkdir(path)
+
+def run_bash(command):
+    exit_code, output = subprocess.getstatusoutput(command)
+    if exit_code != 0:
+        print(output)
+        raise ValueError(f'failed with exit code {exit_code}')
+    return output
+
+class App(object):
 
     def __init__(self):
         parser = argparse.ArgumentParser(description='boomstream.com downloader')
-        parser.add_argument('--url', type=str, required=True)
-        parser.add_argument('--pin', type=str, required=True)
+        parser.add_argument('--entity', type=str, required=True)
+        parser.add_argument('--pin', type=str, required=False)
         parser.add_argument('--use-cache', action='store_true', required=False)
         parser.add_argument('--resolution', type=str, required=False)
         self.args = parser.parse_args()
 
     def get_token(self):
-        if 'records' in self.config['mediaData'] and len(self.config['mediaData']['records']) > 0:
-            return b64decode(self.config['mediaData']['records'][0]['token']).decode('utf-8')
-        else:
-            return b64decode(self.config['mediaData']['token']).decode('utf-8')
+        return b64decode(self.config['mediaData']['token']).decode('utf-8')
```

Comment:
When records are not available yet, token can be retrieved from 'mediaData' directly. The same works for m3u8 playlist.